resources/monitor_diagnostic_settings.go

```diff
@@ -0,0 +1,221 @@
+			},
+			{
+				Name:        "name",
+				Description: "Azure resource name",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "type",
+				Description: "Azure resource type",
+				Type:        schema.TypeString,
+			},
+		},
+		Relations: []*schema.Table{
+			{
+				Name:         "azure_monitor_diagnostic_setting_metrics",
+				Description:  "MetricSettings part of MultiTenantDiagnosticSettings Specifies the settings for a particular metric",
+				Resolver:     fetchMonitorDiagnosticSettingMetrics,
+				AlwaysDelete: true,
+				Columns: []schema.Column{
+					{
+						Name:        "diagnostic_setting_cq_id",
+						Description: "Unique ID of azure_monitor_diagnostic_settings table (FK)",
+						Type:        schema.TypeUUID,
+						Resolver:    schema.ParentIdResolver,
+					},
+					{
+						Name:        "diagnostic_setting_id",
+						Description: "Unique ID of azure_monitor_diagnostic_settings table (FK)",
+						Type:        schema.TypeString,
+						Resolver:    schema.ParentResourceFieldResolver("id"),
+					},
+					{
+						Name:        "time_grain",
+						Description: "the timegrain of the metric in ISO8601 format",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "category",
+						Description: "Name of a Diagnostic Metric category for a resource type this setting is applied to To obtain the list of Diagnostic metric categories for a resource, first perform a GET diagnostic settings operation",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "enabled",
+						Description: "a value indicating whether this category is enabled",
+						Type:        schema.TypeBool,
+					},
+					{
+						Name:        "retention_policy_enabled",
+						Description: "a value indicating whether the retention policy is enabled",
+						Type:        schema.TypeBool,
+						Resolver:    schema.PathResolver("RetentionPolicy.Enabled"),
+					},
+					{
+						Name:        "retention_policy_days",
+						Description: "the number of days for the retention in days A value of 0 will retain the events indefinitely",
+						Type:        schema.TypeInt,
+						Resolver:    schema.PathResolver("RetentionPolicy.Days"),
+					},
+				},
+			},
+			{
+				Name:         "azure_monitor_diagnostic_setting_logs",
+				Description:  "LogSettings part of MultiTenantDiagnosticSettings Specifies the settings for a particular log",
+				Resolver:     fetchMonitorDiagnosticSettingLogs,
+				AlwaysDelete: true,
+				Columns: []schema.Column{
+					{
+						Name:        "diagnostic_setting_cq_id",
+						Description: "Unique ID of azure_monitor_diagnostic_settings table (FK)",
+						Type:        schema.TypeUUID,
+						Resolver:    schema.ParentIdResolver,
+					},
+					{
+						Name:        "diagnostic_setting_id",
+						Description: "Unique ID of azure_monitor_diagnostic_settings table (FK)",
+						Type:        schema.TypeString,
+						Resolver:    schema.ParentResourceFieldResolver("id"),
+					},
```

Comment:
old duplicate of the field above?

Reply:
It is a foreign key to the original id of the resource 
parent -> relation
cq_id -> diagnostic_setting_cq_id
id -> diagnostic_setting_id

Reply:
Ok, I see. What's the benefit of having both fields?

Reply:
I think it adds more flexibility because you can work with original IDs and in some case it can be useful